web/src/stream_ui_updates.js

```diff
@@ -252,7 +252,13 @@ export function enable_or_disable_permission_settings_in_edit_panel(sub) {
 
     const $general_settings_container = $stream_settings.find($("#stream_permission_settings"));
     $general_settings_container
-        .find("input, select, button")
+        .find("input, button")
+        .prop("disabled", !sub.can_change_stream_permissions);
+    const $advanced_configurations_container = $stream_settings.find(
+        $("#stream-advanced-configurations"),
+    );
+    $advanced_configurations_container
+        .find("select, button")
```

Comment:
I think this would also need `input` since message retention days setting can have a custom input box.